src/test/java/uk/gov/justice/digital/hmpps/keyworker/services/NomisBatchServiceTest.kt

```diff
@@ -0,0 +1,105 @@
+package uk.gov.justice.digital.hmpps.keyworker.services
+
+import com.microsoft.applicationinsights.TelemetryClient
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.ArgumentMatchers.any
+import org.mockito.ArgumentMatchers.anyString
+import org.mockito.ArgumentMatchers.eq
+import org.mockito.ArgumentMatchers.isA
+import org.mockito.ArgumentMatchers.isNull
+import org.mockito.Mockito.`when`
+import org.mockito.Mockito.never
+import org.mockito.Mockito.times
+import org.mockito.Mockito.verify
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.mock.mockito.MockBean
+import org.springframework.test.context.ContextConfiguration
+import org.springframework.test.context.junit.jupiter.SpringExtension
+import uk.gov.justice.digital.hmpps.keyworker.config.RetryConfiguration
+import uk.gov.justice.digital.hmpps.keyworker.dto.CaseloadUpdate
+import uk.gov.justice.digital.hmpps.keyworker.dto.Prison
+
+@ExtendWith(SpringExtension::class)
+@ContextConfiguration(classes = [NomisBatchService::class, RetryConfiguration::class])
+class NomisBatchServiceTest {
+  @Autowired
+  private lateinit var batchService: NomisBatchService
+
+  @MockBean
+  private lateinit var nomisService: NomisService
+
+  @MockBean
+  private lateinit var telemetryClient: TelemetryClient
+
+  @Test
+  fun enableNomis_makesPrisonApiCalls() {
+    val prisons = listOf(MDI, LEI, LPI)
+    `when`(nomisService.allPrisons).thenReturn(prisons)
```

Comment:
Should be able to use whenever here. if not might be a good time to add the Kotlin mockito lib